torch/library.py

```diff
@@ -49,9 +49,43 @@ def fallthrough_kernel():
     raise NotImplementedError("fallthrough_kernel() should never be called.")
 
 
-def Schema(name: str, function: Callable, mutates_args=()) -> str:
+def infer_schema(
+    function: Callable, name: Optional[str] = None, mutates_args=()
+) -> str:
+    r"""Parses the schema of a given function with type hints. The schema is inferred from the
+        function's type hints, and can be used to define a new operator.
+
+    We make the following assumptions:
+    - None of the outputs alias any of the inputs or each other.
+    - String type annotations "device, dtype, Tensor, types" without library specification
+      are assumed to be torch.*. Similarly, string type annotations "Optional, List, Sequence, Union"
+      without library specification are assumed to be typing.*.
+    - Only the args listed in ``mutates_args`` are being mutated. If ``mutates_args`` is "unknown",
+        it assumes that all inputs to the operator are being mutates.
+
+    Callers (e.g. the custom ops API) are responsible for checking these assumptions.
+
+    Args:
+        function: function schema to define a new operator.
```

Comment:
function: The function from which to infer a schema for from its type annotations